pins/boards.py

```diff
@@ -654,13 +638,41 @@ def prepare_pin_version(
         versioned: Optional[bool] = None,
         created: Optional[datetime] = None,
         object_name: Optional[str] = None,
+    ):
+        meta = self._create_meta(
+            pin_dir_path,
+            x,
+            name,
+            type,
+            title,
+            description,
+            metadata,
+            versioned,
+            created,
+            object_name,
+        )
+
+        # handle unversioned boards
+        version_setup(self, name, meta.version, versioned)
+
+        return meta
+
+    def _create_meta(
```

Comment:
splitting this out so I can use `_create_meta` for Connect, without running the `version_setup`. Connect needs its own flavor since you cannot delete the latest pin. This causes the following behavior:

Most boards:
make metadata -> check and delete old versions for unversioned -> write new pin

Connect: 
make metadata -> write new pin  ->  check and delete old versions for unversioned